crates/libcontainer/src/network/network_device.rs

```diff
@@ -0,0 +1,238 @@
+use std::fs::File;
+use std::os::fd::AsRawFd;
+use std::path::Path;
+
+use netlink_packet_route::address::{AddressAttribute, AddressFlags, AddressMessage, AddressScope};
+use oci_spec::runtime::LinuxNetDevice;
+
+use super::Result;
+use super::address::AddressClient;
+use super::link::LinkClient;
+use super::wrapper::create_network_client;
+use crate::network::serialize::SerializableAddress;
+
+/// dev_change_netns allows to move a device given by name to a network namespace given by nsPath
+/// and optionally change the device name.
+/// The device name will be kept the same if device.Name is None or an empty string.
+/// This function ensures that the move and rename operations occur atomically.
+/// It preserves existing interface attributes, including IP addresses.
+pub fn dev_change_net_namespace(
+    name: &str,
+    netns_path: &Path,
+    device: &LinuxNetDevice,
+) -> Result<Vec<SerializableAddress>> {
+    tracing::debug!(
+        "attaching network device {} to network namespace {}",
+        name,
+        netns_path.display()
+    );
+
+    let mut link_client = LinkClient::new(create_network_client())?;
+    let mut addr_client = AddressClient::new(create_network_client())?;
+
+    let netns_file = File::open(netns_path)?;
```

Comment:
Please pass `netns_file` to dev_change_net_namespace() because the caller may want to validate whether the netns_path is real or not.